ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OKMeansTest.scala

```
@@ -0,0 +1,121 @@
+    val algo = new H2OKMeans()
+      .setSplitRatio(0.8)
+      .setSeed(1)
+      .setK(3)
+      .setFeaturesCols("sepal_len", "sepal_wid", "petal_len", "petal_wid")
+
+    val model = algo.fit(dataset)
+    val transformed = model.transform(dataset)
+    assert(transformed.select("prediction").head() == Row(0))
+    assert(transformed.select("prediction").distinct().count() == 3)
+  }
+
+  test("H2OKMeans' full result in prediction details column") {
+    val algo = new H2OKMeans()
+      .setSplitRatio(0.8)
+      .setSeed(1)
+      .setK(3)
+      .setWithDetailedPredictionCol(true)
+      .setFeaturesCols("sepal_len", "sepal_wid", "petal_len", "petal_wid")
+
+    val model = algo.fit(dataset)
+    val transformed = model.transform(dataset)
+    assert(transformed.select("detailed_prediction.cluster").head().getInt(0) == 0)
+    assert(transformed.select("detailed_prediction.distances").head().getAs[Seq[Double]](0).length == 3)
+  }
+
+  test("H2OKMeans user points") {
+    val algo = new H2OKMeans()
+      .setSplitRatio(0.8)
+      .setSeed(1)
+      .setK(3)
+      .setUserPoints(Array(Array(4.9, 3.0, 1.4, 0.2), Array(5.6, 2.5, 3.9, 1.1), Array(6.5, 3.0, 5.2, 2.0)))
+      .setFeaturesCols("sepal_len", "sepal_wid", "petal_len", "petal_wid")
+
+    val model = algo.fit(dataset)
+    assert(model.transform(dataset).select("prediction").head() == Row(0))
+  }
+
+  test("H2OKMeans single string column input without converting to categorical") {
+    val algo = new H2OKMeans()
+      .setSplitRatio(0.8)
+      .setSeed(1)
+      .setK(3)
+      .setFeaturesCols("class")
+      .setAllStringColumnsToCategorical(false)
+
+    intercept[H2OModelBuilderIllegalArgumentException] {
+      algo.fit(dataset)
+    }
+  }
+
+  test("H2OKMeans string column input with other non-string column") {
+    val algo = new H2OKMeans()
+      .setSplitRatio(0.8)
+      .setSeed(1)
+      .setK(3)
+      .setFeaturesCols("class", "sepal_wid")
```

I assume that `class` column is ignored. Shouldn't we return an exception as in the previous test and let a user to specify a proper list of feature columns?

---

Agree